pkg/kubelet/kubelet_pods.go

```diff
@@ -1885,6 +1886,13 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			ImageID:      cs.ImageID,
 			ContainerID:  cid,
 		}
+		if !cs.StartedAt.IsZero() && cs.FinishedAt.IsZero() && !hasStartupProbe {
```

Comment:
I think we have to consider `cs.State.Running != nil` as well because `startedAt` field is not cleared after the container completes.

Reply:
But `FinishedAt` should be there to tell, no?

Reply:
Ah... I think you are right.